controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java

```diff
@@ -28,89 +31,47 @@
+        maintainers.add(new ApplicationOwnershipConfirmer(controller, intervals.applicationOwnershipConfirmer, controller.serviceRegistry().ownershipIssues()));
+        maintainers.add(new SystemUpgrader(controller, intervals.systemUpgrader));
+        maintainers.add(new JobRunner(controller, intervals.jobRunner));
+        maintainers.add(new OsVersionStatusUpdater(controller, intervals.defaultInterval));
+        maintainers.add(new ContactInformationMaintainer(controller, intervals.contactInformationMaintainer));
+        maintainers.add(new NameServiceDispatcher(controller, intervals.nameServiceDispatcher));
+        maintainers.add(new CostReportMaintainer(controller, intervals.costReportMaintainer, controller.serviceRegistry().costReportConsumer()));
+        maintainers.add(new ResourceMeterMaintainer(controller, intervals.resourceMeterMaintainer, metric, controller.serviceRegistry().meteringService()));
+        maintainers.add(new CloudEventReporter(controller, intervals.cloudEventReporter, metric));
+        maintainers.add(new RotationStatusUpdater(controller, intervals.defaultInterval));
+        maintainers.add(new ResourceTagMaintainer(controller, intervals.resourceTagMaintainer, controller.serviceRegistry().resourceTagger()));
+        maintainers.add(new SystemRoutingPolicyMaintainer(controller, intervals.systemRoutingPolicyMaintainer));
+        maintainers.add(new ApplicationMetaDataGarbageCollector(controller, intervals.applicationMetaDataGarbageCollector));
+        maintainers.add(new ContainerImageExpirer(controller, intervals.containerImageExpirer));
+        maintainers.add(new HostSwitchUpdater(controller, intervals.hostSwitchUpdater));
+        maintainers.add(new ReindexingTriggerer(controller, intervals.reindexingTriggerer));
     }
 
     public Upgrader upgrader() { return upgrader; }
 
     @Override
     public void deconstruct() {
-        deploymentExpirer.close();
-        deploymentIssueReporter.close();
-        metricsReporter.close();
-        outstandingChangeDeployer.close();
-        versionStatusUpdater.close();
-        upgrader.close();
-        readyJobsTrigger.close();
-        deploymentMetricsMaintainer.close();
-        applicationOwnershipConfirmer.close();
-        systemUpgrader.close();
-        osUpgraders.forEach(ControllerMaintainer::close);
-        osVersionStatusUpdater.close();
-        jobRunner.close();
-        contactInformationMaintainer.close();
-        costReportMaintainer.close();
-        resourceMeterMaintainer.close();
-        nameServiceDispatcher.close();
-        cloudEventReporter.close();
-        rotationStatusUpdater.close();
-        resourceTagMaintainer.close();
-        systemRoutingPolicyMaintainer.close();
-        applicationMetaDataGarbageCollector.close();
-        containerImageExpirer.close();
-        hostSwitchUpdater.close();
+        maintainers.forEach(Maintainer::shutdown);
+        maintainers.forEach(Maintainer::close);
```

Comment:
Better, but why do we need both if close calls shutdown when necessary?

Reply:
If it's to initiate shutdown on everything at once and then wait, I suggest renaming `close`. `awaitShutdown`?

Reply:
Good idea.